core/src/main/java/com/linecorp/armeria/internal/server/RouteDecoratingService.java

```diff
@@ -85,9 +85,9 @@ public static Function<? super HttpService, InitialDispatcherService> newDecorat
     private final Route route;
     private final HttpService decorator;
 
-    public RouteDecoratingService(Route route,
+    public RouteDecoratingService(Route route, String contextPath,
                                   Function<? super HttpService, ? extends HttpService> decoratorFunction) {
-        this.route = requireNonNull(route, "route");
+        this.route = requireNonNull(route, "route").withPrefix(contextPath);
```

Comment:
This is not your fault, but it seems like a `DefaultRoute` is created even if the `contextPath` is `/`. Let's fix `DefaultRoute.withPrefix()` method as well.

Reply:
Let me handle this in a separate PR 🙇

Reply:
https://github.com/line/armeria/pull/5235
Created a pull request here